app/grandchallenge/algorithms/forms.py

```diff
@@ -365,21 +365,77 @@ class Meta:
             "workstation_config": "Viewer Configuration",
         }
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, *args, user, **kwargs):
+        super().__init__(*args, user=user, **kwargs)
+        self._user = user
 
         self.fields["contact_email"].required = True
         self.fields["display_editors"].required = True
 
-        self.fields["job_requires_gpu_type"].choices = [
-            (choice.value, choice.label)
-            for choice in [GPUTypeChoices.NO_GPU, GPUTypeChoices.T4]
-        ]
+        self.fields["job_requires_gpu_type"].choices = (
+            self.get_selectable_gpu_type_choices()
+        )
         self.fields["job_requires_memory_gb"].validators = [
             MinValueValidator(settings.ALGORITHMS_MIN_MEMORY_GB),
-            MaxValueValidator(settings.ALGORITHMS_MAX_MEMORY_GB),
+            MaxValueValidator(self.get_maximum_settable_memory_gb()),
         ]
 
+    @cached_property
```

Comment:
I don't think that caching will help here as it is returning an unevaluated queryset.

Reply:
I now removed cached from the decorator for this reason. I also looked into whether we could combine the two queries into a single one using annotations. But that ultimately ended up being slower, since I would have to loop through the resulting annotated queryset in python. The current code with two separate queries is also much easier to read. So I left it with two separate queries.

Reply:
This is still a `cached_property` and would it be faster to first filter the queryset by `public` and `submission_kind` before annotating the counts?

Reply:
When I inspect the executed SQL, they are the same regardless of where I place the filter for `public` and `submission_kind`. Apparently there is some optimization going on under the hood.